### PR TITLE
Refine some comments about learner read

### PR DESCRIPTION
**Pull request by JaySon-Huang**
Signed-off-by: JaySon-Huang <jayson.hjs@gmail.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflash/issues/4118

Problem Summary: During refactoring, we may change some behavior unexpectedly https://github.com/pingcap/tiflash/pull/4674. Add some comments about learner read to explain why those steps are important.

### What is changed and how it works?

* Add comments about the order of learner read and table structure lock
* Make some member vars to be local vars, avoid too much internal state changing for DAGStorageInterpreter

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by JaySon-Huang** [comment]
/hold
pending for https://github.com/pingcap/tiflash/pull/4784 to be merged first

**Review by SeaRise — APPROVED** [review]

File: dbms/src/Functions/GeoUtils.h [hunk 1]
```diff
@@ -30,6 +30,7 @@
 #endif
 
 #pragma GCC diagnostic ignored "-Wpragmas"
+#pragma GCC diagnostic ignored "-Wunknown-warning-option"
```
Comment (JaySon-Huang) [inline dbms/src/Functions/GeoUtils.h:33]: This is to fix compile error under MacOS with Apple clang version 13.0.0

**Comment by JaySon-Huang** [comment]
@SeaRise @windtalker @ywqzzy PTAL again

File: dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp [hunk 2]
```diff
@@ -233,13 +240,19 @@ void DAGStorageInterpreter::execute(DAGPipeline & pipeline)
...
-    buildRemoteRequests();
+    auto remote_requests = buildRemoteRequests();
 
+    // A failpoint to test pause before alter lock released
+    FAIL_POINT_PAUSE(FailPoints::pause_with_alter_locks_acquired);
+    // Release alter locks
+    // The DeltaTree engine ensures that once input streams are created, the caller can get a consistent result
+    // from those streams even if DDL operations are applied. Release the alter lock so that reading does not
+    // block DDL operations, keep the drop lock so that the storage not to be dropped during reading.
     releaseAlterLocks();
```
Comment (SeaRise) [inline dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp:256]:
```suggestion
    const auto drop_locks = releaseAlterLocks();
```

File: dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp [hunk 3]
```diff
@@ -954,8 +990,10 @@ std::tuple<Names, NamesAndTypes, std::vector<ExtraCastAfterTSMode>> DAGStorageIn
     return {required_columns_tmp, source_columns_tmp, need_cast_column};
 }
 
-void DAGStorageInterpreter::buildRemoteRequests()
+// Build remote requests from `region_retry_from_local_region`
```
Comment (SeaRise) [inline dbms/src/Flash/Coprocessor/DAGStorageInterpreter.cpp:993]:
```suggestion
// Build remote requests from `region_retry_from_local_region` and `table_regions_info.remote_regions`
```

**Review by SeaRise — APPROVED** [review]
other lgtm

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**